packages/app-store/deel/api/add.ts

```diff
@@ -1,20 +1,49 @@
-import { createDefaultInstallation } from "@calcom/app-store/_utils/installation";
-import type { AppDeclarativeHandler } from "@calcom/types/AppHandler";
-
-import appConfig from "../config.json";
-
-const handler: AppDeclarativeHandler = {
-  appType: appConfig.type,
-  variant: appConfig.variant,
-  slug: appConfig.slug,
-  supportsMultipleInstalls: false,
-  handlerType: "add",
-  redirect: {
-    newTab: true,
-    url: "https://go.cal.com/deel",
-  },
-  createCredential: ({ appType, user, slug, teamId }) =>
-    createDefaultInstallation({ appType, user: user, slug, key: {}, teamId }),
-};
-
-export default handler;
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { throwIfNotHaveAdminAccessToTeam } from "@calcom/app-store/_utils/throwIfNotHaveAdminAccessToTeam";
+import prisma from "@calcom/prisma";
+
+import config from "../config.json";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!req.session?.user?.id) {
+    return res.status(401).json({ message: "You must be logged in to do this" });
+  }
+
+  const { teamId } = req.query;
+
+  await throwIfNotHaveAdminAccessToTeam({ teamId: Number(teamId) ?? null, userId: req.session.user.id });
+  const installForObject = teamId ? { teamId: Number(teamId) } : { userId: req.session.user.id };
```

Comment:
there should be null check for `session.user.id`.

Reply:
I think the check at lines 9-11 is for that purpose